core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -99,12 +110,23 @@ public int read() throws IOException {
     if (mPosition == mLength) { // at end of file
       return -1;
     }
-    updateStream();
-    int result = mBlockInStream.read();
-    if (result != -1) {
-      mPosition++;
+    CountingRetry retry = new CountingRetry(MAX_WORKERS_TO_RETRY);
+    IOException lastException = null;
+    while (retry.attempt()) {
```

Comment:
@bf8086 Please check if this still correct.

Reply:
LGTM.